### ISSUE TITLE
Client: Create file for reading

### ISSUE THREAD

**Issue by jlee-r7** [rapid7/smb2#14]
something like this:
```ruby
tree_id = client.tree_connect("\\\\hostname\\share")
client.read(tree_id, "file.txt")
# => "contents of file.txt"
```
needs to 
- send a `CreateRequest` with appropriate values
- read the response, store the `file_id` therein
- send a `GetInfo` request for that `file_id`
- read the response, store the size
- send a `ReadRequest` for that many bytes
- return the value as a String

**Comment by limhoff-r7** [comment]
Is the `tree_id` valid on other clients or is it client specific?  If it's client specific, wouldn't it make more sense for `client.tree_connect` to return a `Tree` object and have it have `Tree.read` as the client is implied by the returned `tree`.

**Comment by jlee-r7** [comment]
Oh, that's a good idea.  Yes, tree IDs are client-specific and Create will be tree-specific.

**Comment by jlee-r7** [comment]
Current thinking looks like this:
```ruby
tree = client.tree_connect("\\\\hostname\\share") # => #<Smb2::Tree ... >
file = tree.create("\\path\\to\\file.txt") # => #<Smb2::File ... >
file.read # => "contents of file.txt"
```

**Comment by jlee-r7** [comment]
fixed in #28
